### PR TITLE
New settings

### LINKED ISSUE #534: Get default post privacy from the API
Instead of having to specify default post privacy in the settings, Tusky should instead fetch it from the API - https://github.com/tootsuite/documentation/blob/master/Using-the-API/API.md#getting-the-current-user

### LINKED ISSUE #607: Adding a per user "show sensitive content" option
Extending issue #385, I wanted to suggest an option, to automatically show "hidden content" (see for example [this account](https://high.cat/@archillect)), _but_ on a per-user basis, since there are certain accounts I know won't disturb me, but use this option on all their posts anyways, and others which most certainly do, which I would very much want to hide.

### LINKED ISSUE #716: Introducing per-account settings
Currently, only the notification settings can be applied per-account, all other settings have a global value.

In the next releases I want to introduce more settings like that
- always show sensitive content (see #607)
- default post privacy (will sync with mastodon web) (see #534)
- mark images as sensitive per default? (will sync with mastodon web)
- maybe an possibility to easier see which account is the currently logged in (see #670)

So the question is where should I put the settings? I see three possibilities:
A) Leave them in the settings. Will let to a lot of settings eventually. (vote with 👍)
B) Put them in the "Edit prodile" page. Page may get very long, it is still missing support for account fields. (vote with 😄)
C) Add a new entry in the sidemenu. Not sure if it will be used often enough to justify a new entry. (vote with 🎉)

All possibilities will probably need some kind of explanation or visual indication, which setting will sync with mastodon web and which wont, otherwise that may confuse users a lot.

### PR DESCRIPTION
**Pull request by connyduck**
new settings, as outlined in https://github.com/tuskyapp/Tusky/issues/716

closes #716, closes #607, closes #534 

  

- new menu entry "account preferences", blocked and muted users have moved
- its now possible to set more options per account, and some will even sync with mastodon server
- new option "always mark media as sensitive", default off
- finally all fragments inherit from `android.support.v4.app.Fragment` so injection works everywhere. had to change the emoji preference and the regex preference for this

note that the indentation of the preference category title is a bug in the Android support library that will be fixed in their next release

not sure about the new icon for Drafts. Another possibility would be to use something like that http://materialdesignicons.com/icon/book-multiple-variant

todo:
- db migrations
- clean up and convert more stuff to Kotlin
- test

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/fragment/NotificationPreferencesFragment.kt [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            soundPref.isChecked = activeAccount.notificationSound
+            soundPref.onPreferenceChangeListener = this
+
+            val vibrationPref = findPreference("notificationAlertVibrate") as SwitchPreference
+            vibrationPref.isChecked = activeAccount.notificationVibration
+            vibrationPref.onPreferenceChangeListener = this
+
+            val lightPref = findPreference("notificationAlertLight") as SwitchPreference
+            lightPref.isChecked = activeAccount.notificationLight
+            lightPref.onPreferenceChangeListener = this
```
Comment (Tak) [inline app/src/main/java/com/keylesspalace/tusky/fragment/NotificationPreferencesFragment.kt]: That's a lot of boilerplate 🙂

File: app/src/main/java/com/keylesspalace/tusky/db/Converters.kt [hunk 2]
```diff
@@ -33,4 +34,14 @@ class Converters {
     fun emojiListToJson(emojiList: List<Emoji>?): String {
         return gson.toJson(emojiList)
     }
+
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/db/Converters.kt]: I have very similar converter in #892 
Maybe we can extract it later

File: app/src/main/res/drawable/ic_account_settings.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
...25 15.1512,15.63L14.15,17.3587C14.0913,17.4713 14.1112,17.5975 14.2087,17.6812L15.2688,18.5113C15.2487,18.6712 15.2388,18.8275 15.2388,18.9988C15.2388,19.17 15.2487,19.3312 15.2688,19.4925L14.2087,20.3225C14.1212,20.4 14.0913,20.5325 14.15,20.64L15.1512,22.3687C15.21,22.48 15.3413,22.52 15.4587,22.48L16.6988,21.9825C16.9575,22.1825 17.2412,22.3488 17.5487,22.4713L17.7388,23.7888C17.7588,23.9113 17.8612,23.9988 17.9887,23.9988L19.99,23.9988C20.1123,23.9988 20.22,23.9113 20.2393,23.7888L20.4297,22.4713C20.7275,22.3387 21.0205,22.1825 21.27,21.9825L22.52,22.48C22.6312,22.52 22.7588,22.48 22.8325,22.3687L23.8288,20.64C23.8913,20.5325 23.8575,20.4 23.77,20.3225L22.7002,19.4925C22.72,19.3312 22.7393,19.17 22.7393,18.9988C22.7393,18.8275 22.7295,18.6712 22.7002,18.5113L23.76,17.6812C23.8475,17.5975 23.8812,17.4713 23.8188,17.3587L22.8225,15.63C22.7588,15.5225 22.6312,15.4787 22.51,15.5175L21.27,16.02C21.0107,15.82 20.7275,15.65 20.42,15.5275L20.2295,14.2087C20.22,14.0912 20.1123,13.9987 19.99,13.9987M18.9888,17.5C19.8188,17.5 20.4888,18.1687 20.4888,18.9988C20.4888,19.8288 19.8188,20.4975 18.9888,20.4975C18.1588,20.4975 17.49,19.8288 17.49,18.9988C17.49,18.1687 18.1588,17.5 18.9888,17.5Z"
```
Comment (charlag) [inline app/src/main/res/drawable/ic_account_settings.xml]: This is some complex SVG
Is this a cog?

File: app/src/main/java/com/keylesspalace/tusky/fragment/NotificationPreferencesFragment.kt (hunk 1 above)
Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/fragment/NotificationPreferencesFragment.kt]: We can invent a function or use something like `apply` but it doesn't worth it I guess

**Comment by charlag** [comment]
This looks really sick!
If you need any help with migrations or testing migrations, don't hesitate to ping me :wink:

File: app/src/main/res/drawable/ic_account_settings.xml (hunk 3 above)
Comment (connyduck) [reply app/src/main/res/drawable/ic_account_settings.xml]: its this icon https://github.com/Templarian/MaterialDesign/blob/master/icons/svg/account-settings-variant.svg

**Comment by connyduck** [comment]
So I think this is now done! Would be great if somebody else could test it before its merged!

**Comment by charlag** [comment]
I will try to check it today or tomorrow

**Comment by Tak** [comment]
Looks good to me, I'll run it on my phone for a while

**Comment by Tak** [comment]
Did you want to remove the setting for notification check interval while you're in there?

**Comment by connyduck** [comment]
> Did you want to remove the setting for notification check interval while you're in there?

no, because feedback was very mixed I decided to leave it

File: app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.java [hunk 4]
```diff
@@ -244,9 +248,8 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
 
     private void setupTimelinePreferences() {
         SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(getActivity());
-        preferences.registerOnSharedPreferenceChangeListener(this);
-        alwaysShowSensitiveMedia = preferences.getBoolean("alwaysShowSensitiveMedia", false);
-        boolean mediaPreviewEnabled = preferences.getBoolean("mediaPreviewEnabled", true);
+        alwaysShowSensitiveMedia = accountManager.getActiveAccount().getMediaPreviewEnabled();
```
Comment (Tak) [inline app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.java]: This seems problematic :wink:

Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.java]: yes indeed, thx for noticing

**Comment by ariasuni** [comment]
Awesome, thanks a lot!!

**Pull request merged**